src/ngx_http_lua_module.c

```diff
@@ -972,6 +972,16 @@ ngx_http_lua_merge_srv_conf(ngx_conf_t *cf, void *parent, void *child)
 
 #   endif
 
+
+#   ifdef TLSEXT_TYPE_application_layer_protocol_negotiation
+        SSL_CTX_set_alpn_select_cb(sscf->ssl.ctx, ngx_http_lua_ssl_alpn_select, NULL);
```

Comment:
Hmm, maybe we should do `alpn_by_lua_block` and `alpn_by_lua_file` directives by allowing running Lua directly in the APLN and NPN hooks? This way we can ensure our application level protocol selection works on all the code paths (ssl session id and tls session ticket) in case of SSL session resumption.

Reply:
Sorry for my fault. I don't have the test session. I'm starting to change to it. You can close related PR.